### DISCUSSION TITLE
SWC migration path

### DISCUSSION THREAD

**Discussion by TxHawks** [vercel/next.js#28130] [Help] [answered]
Hello 👋

With the coming planned switch from Babel to SWC, I'm wondering if there is an RFC or a planned migration path for projects that use custom babel plugins that aren't supported by SWC, such as `babel-plugin-macros` or custom transforms.

I tried searching through the issues and discussions but couldn't find anything relevant

**Comment by NateRadebaugh** [comment]
My understanding is that only the code framework pieces are compiled using SWC. There's no plan to move away from webpack + babel for our apps.

**Reply by timneutkens** [reply]
You can read the blogpost here: https://nextjs.org/blog/next-11-1#adopting-rust-based-swc.

**Reply by NateRadebaugh** [reply]
Thanks for clarifying, I must have missed this blog post.

**Comment by timneutkens** [comment, marked as answer]
If you have custom Babel configuration you'll be using Babel still, so it's recommended to not use anything that does not absolutely require custom Babel configuration. We're currently working on having full compatibility with what Next.js has by default including all custom transforms Next.js has. Once that is shipped we'll figure out which Babel plugins are needed for ecosystem compat, e.g. styled-components and such might have to be ported.

**Reply by TxHawks** [reply]
Thanks for the clarification.

Will it be possible to use Babel for transpiling code and SWC for minifying the output?

**Reply by timneutkens** [reply]
Yeah definitely, in the future that will be enabled by default given that it is not affected by user configuration 👍

**Comment by ghost** [comment]
how to use the SWC in my project
